src/OrleansPSUtils/StartGrainClient.cs

```diff
@@ -0,0 +1,92 @@
+﻿using Orleans;
+using Orleans.Runtime.Configuration;
+using System;
+using System.IO;
+using System.Management.Automation;
+using System.Net;
+
+namespace OrleansPSUtils
+{
+    [Cmdlet(VerbsLifecycle.Start, "GrainClient", DefaultParameterSetName = DefaultSet)]
+    public class StartGrainClient : PSCmdlet
+    {
+        private const string DefaultSet = "Default";
+        private const string FilePathSet = "FilePath";
+        private const string FileSet = "File";
+        private const string ConfigSet = "Config";
+        private const string EndpointSet = "Endpoint";
+
+        [Parameter(Position = 1, Mandatory = true, ValueFromPipeline = true, ParameterSetName = FilePathSet)]
+        public string ConfigFilePath { get; set; }
+
+        [Parameter(Position = 2, Mandatory = true, ValueFromPipeline = true, ParameterSetName = FileSet)]
+        public FileInfo ConfigFile { get; set; }
+
+        [Parameter(Position = 3, Mandatory = true, ValueFromPipeline = true, ParameterSetName = ConfigSet)]
+        public ClientConfiguration Config { get; set; }
+
+        [Parameter(Position = 4, Mandatory = true, ValueFromPipeline = true, ParameterSetName = EndpointSet)]
+        public IPEndPoint GatewayAddress { get; set; }
```

Comment:
The position behaves differently when you have `ParameterSet`s. Basically it follow the order inside the own ParameterSet so yes, you will not pass more than 1 parameter for the same set. They are made by powershell runtime and you can see it on `Do-Something -?` and it will show you the possible sets acceptable.